### PR TITLE
fixes for finalized TLFs CORE-4514

### PR DESCRIPTION
**Pull request by mmaxim**
Patch does a bunch of things:

1.) Makes it possible to read from finalized TLFs that are superseded by changing inbox.go to allow them to be returned if the `ConvID` param is set.
2.) Render message errors cleaner in the CLI.
3.) Render sender username in finalized convs.
4.) Return finalized superseded convs from the JSON API, add ability for JSON API to return all hats on a TLF (even finalized ones).
5.) Add ability to JSON API to display device revoked status.
6.) Label all messages from a reset user as from a revoked device.

### CODE REVIEW

File: go/chat/boxer.go [hunk 1]
```diff
@@ -563,9 +575,11 @@ func (b *Boxer) ValidSenderKey(ctx context.Context, sender gregor1.UID, key []by
 	}
 
 	if deleted {
-		// XXX something else to flag as a deleted key?
-		b.log().Warning("sender %s key %s was deleted", kbSender, kid)
-		return true, true, nil, nil
+		b.Debug(ctx, "sender %s key %s was deleted", kbSender, kid)
+		// Set the key as being revoked since the beginning of time, so all messages will get labeled
+		// as suspect
+		zeroTime := gregor1.Time(0)
```
Comment (mmaxim) [inline go/chat/boxer.go]: This is a key change, we talked about it earlier, but I figured I'd point it out.

File: go/client/chat_cli_rendering.go [hunk 2]
```diff
@@ -182,23 +182,25 @@ func (v conversationListView) show(g *libkb.GlobalContext, myUsername string, sh
...
+					msg = &mCopy
 				}
 			}
 		}
 		if msg == nil {
-			// Skip conversations with no visible messages.
-			// This should never happen.
-			g.Log.Error("Skipped conversation with no visible messages: %s", conv.Info.Id)
-			continue
+			// Make a fake error in case we have a non-empty thread with no visible message
```
Comment (mmaxim) [inline go/client/chat_cli_rendering.go]: Without this, if the latest message is an error, we just don't render the row.

File: go/client/chat_svc_handler.go [hunk 3]
```diff
@@ -54,9 +54,10 @@ func (c *chatServiceHandler) ListV1(ctx context.Context, opts listOptionsV1) Rep
 
 	res, err := client.GetInboxAndUnboxLocal(ctx, chat1.GetInboxAndUnboxLocalArg{
 		Query: &chat1.GetInboxLocalQuery{
-			Status:     utils.VisibleChatConversationStatuses(),
-			TopicType:  &topicType,
-			UnreadOnly: opts.UnreadOnly,
+			Status:            utils.VisibleChatConversationStatuses(),
+			TopicType:         &topicType,
+			UnreadOnly:        opts.UnreadOnly,
+			OneChatTypePerTLF: new(bool),
```
Comment (mmaxim) [inline go/client/chat_svc_handler.go]: Need this because we want to include all convos in this interface.

File: go/chat/boxer.go [hunk 4]
```diff
@@ -106,9 +106,13 @@ func (b *Boxer) UnboxMessage(ctx context.Context, boxed chat1.MessageBoxed, fina
 
 	username, deviceName, deviceType, err := b.getSenderInfoLocal(ctx, pt.ClientHeader)
 	if err != nil {
-		b.log().Warning("unable to fetch sender informaton: UID: %s deviceID: %s",
+		b.Debug(ctx, "unable to fetch sender and device informaton: UID: %s deviceID: %s",
 			pt.ClientHeader.Sender, pt.ClientHeader.SenderDevice)
-		// ignore non-fatal error
+		// try to jsut get username
```
Comment (patrickxb) [inline go/chat/boxer.go]: just

File: go/chat/boxer.go (hunk 1 above)
Comment (patrickxb) [inline go/chat/boxer.go]: yup

**Pull request merged**